sources/string/parser.cpp

```diff
@@ -55,45 +56,64 @@
 //         core::Path::getFilePathUnsafe(aFolderName, aFilename));
 // }
 
-std::vector<str::string>
-str::Parser::slice(const str::string& aStr,
+std::vector<std::string_view>
+str::Parser::slice(const std::string_view& aStr,
                    const str::string& aDelimiters,
                    const str::string& aErase) noexcept
 {
-    std::vector<str::string> result(1);
+    std::vector<std::string_view> result;
+    size_t start    = 0;
+    size_t char_pos = 0;
+
+    auto is_erase_char = [&](char c)
+    { return aErase.find(c) != str::string::npos; };
+
+    auto is_delimiter_char = [&](char c)
+    { return aDelimiters.find(c) != str::string::npos; };
 
-    for (auto i : aStr)
+    for (char c : aStr)
     {
-        if (aErase.find(i) != str::string::npos)
+        if (is_erase_char(c))
         {
             continue;
```

Comment:
`std::string_view` is a continuous stream of characters. You must make changes to `aStr` to meet this criterion by using two pointers.